ssim/ssimlib.py

```diff
@@ -78,7 +78,11 @@ def ssim_value(self, img2):
         Returns:
           Computed SSIM float value.
         """
-        img2 = SSIMImage(img2, self.gaussian_kernel_1d, self.img1.size)
+        # Performance boost if handed a compatible SSIMImage object.
+        if not isinstance(img2, SSIMImage) \
+          or not (self.gaussian_kernel_1d == img2.gaussian_kernel_1d).all():
```

Comment:
I did some digging around on NumPy and found this well written post on Stack Overflow:

http://stackoverflow.com/questions/10580676/comparing-two-numpy-arrays-for-equality

It reveals that (A==B).all() can be tripped up in certain circumstances, like if A or B is empty, so I'm going to switch to the more robust np.array_equal(A,B) solution.